src/main/java/de/tum/in/www1/artemis/service/TextSubmissionService.java

```diff
@@ -143,6 +146,9 @@ public TextSubmission save(TextSubmission textSubmission) {
      */
     @Transactional(readOnly = true)
     public Optional<TextSubmission> getTextSubmissionWithoutManualResult(TextExercise textExercise) {
+        if (textExercise.isAutomaticAssessmentEnabled() && textAssessmentQueueService.isPresent()) {
+            return textAssessmentQueueService.get().getProposedTextSubmission(textExercise);
```

Comment:
Can we also pass a preferred language here and try to consider it if possible?

Reply:
Done